### PR TITLE
More tests 3

### PR DESCRIPTION
**Pull request by MajorBreakfast**

### CODE REVIEW

File: futures-util/src/sink/map_err.rs [hunk 1]
```diff
@@ -59,19 +59,19 @@ impl<S, F, E> Sink for SinkMapErr<S, F>
...
     }
 
     fn poll_flush(mut self: PinMut<Self>, cx: &mut task::Context) -> Poll<Result<(), Self::SinkError>> {
-        self.sink().poll_flush(cx).map_err(self.expect_f())
+        self.sink().poll_flush(cx).map_err(|e| self.expect_f()(e))
     }
 
     fn poll_close(mut self: PinMut<Self>, cx: &mut task::Context) -> Poll<Result<(), Self::SinkError>> {
-        self.sink().poll_close(cx).map_err(self.expect_f())
+        self.sink().poll_close(cx).map_err(|e| self.expect_f()(e))
```
Comment (MajorBreakfast) [inline futures-util/src/sink/map_err.rs]: Why did this fail?

Comment (cramertj) [reply futures-util/src/sink/map_err.rs]: Ah-- this wasn't the same. `expect_f` is only called in the case of an error in this version. With the clippy-suggested change, `expect_f` is called regardless of whether an error occurs or not.

Comment (MajorBreakfast) [reply futures-util/src/sink/map_err.rs]: Ah I see. It‘s not a method created by the unsafe_unpinned macro. Good that we have unit tests :)

Comment (cramertj) [reply futures-util/src/sink/map_err.rs]: :)

**Pull request merged**